### PR TITLE
Avoid using the default logger in the graph lib and optimizers

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

1. Use the session logger if it is available. 
2. Don't disable warning 4100 globally. We should fix the warnings instead of disabling it.

**Motivation and Context**
- Why is this change required? What problem does it solve?

Use the session logger if we can.

- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by skottmckay** [comment]
What's the reason for wanting to remove the default logger? What will be used for logging outside of a session (e.g. failure during startup)?

**Comment by snnn** [comment]
> What's the reason for wanting to remove the default logger? What will be used for logging outside of a session (e.g. failure during startup)?

If it's a failure, we can return Status. And ORT shouldn't have any failure before creating the InferenceSession. For the code I just changed, they all happen after that. As the session logger is already created and available, we should use it.

File: include/onnxruntime/core/graph/node_arg.h [hunk 1]
```diff
@@ -68,13 +68,13 @@ class NodeArg {
   @param strict If true, the shape update will fail if there are incompatible values. 
                 If false, will be lenient and merge only shape info that can be validly processed.
   @returns Success unless there is existing type or shape info that can't be successfully updated. */
-  common::Status UpdateTypeAndShape(const ONNX_NAMESPACE::TypeProto& input_type, bool strict = true);
+  common::Status UpdateTypeAndShape(const ONNX_NAMESPACE::TypeProto& input_type, bool strict, const logging::Logger* logger);
```
Comment (skottmckay) [inline include/onnxruntime/core/graph/node_arg.h]: >const logging::Logger* logger [](start = 94, length = 29)

Shouldn't we enforce that there is always a logger available? Much simpler than having 'if (logger != nullptr)' checks everywhere.

Comment (snnn) [reply include/onnxruntime/core/graph/node_arg.h]: Thanks. But that requires a big change.  I'm doing it.

File: onnxruntime/test/framework/shape_inference_test.cc [hunk 2]
```diff
@@ -22,7 +24,7 @@ class ShapeInferenceTest : public ::testing::Test {
   std::unordered_map<string, std::unique_ptr<onnxruntime::NodeArg>> name_to_arg_;
 
  public:
-  ShapeInferenceTest() : model_("Test"), node_count_(0) {}
+  ShapeInferenceTest() : model_("Test", false, ::onnxruntime::test::DefaultLoggingManager().DefaultLogger()), node_count_(0) {}
```
Comment (skottmckay) [inline onnxruntime/test/framework/shape_inference_test.cc]: >::onnxruntime [](start = 47, length = 13)

nit: pet peeve. using '::' in front of onnxruntime should never be necessary

**Comment by snnn** [comment]
Scott is right, I will not remove the default logger.  I'll keep it there in onnxruntime_common.

**Pull request merged**